### DISCUSSION TITLE
[Feature] Support UnifiedPush (ntfy, Google's FCM, etc) for push notifications

### DISCUSSION THREAD

**Discussion by aero31aero** [immich-app/immich#12842] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This is the one missing link, I believe, that'd take Immich from a good app for shared albums, to a great app for shared albums.

If https://unifiedpush.org/ spec is implemented, people would be able to use their own self-hosted ntfy instances, or some other method of their own choosing, to receive push notifications in a self-hosted fashion.

I'm pretty sure the devs already know of it, but for others, here's what I'm talking about (from the FAQ section of the above website):

![image](https://github.com/user-attachments/assets/ce7aacc4-4669-4e87-bd78-69b68b6a9e95)

- Immich Server will send notifications to your self-hosted, or some other publicly hosted, push server.
- Your push notification application will maintain an open connection to that public server at all times.
- On getting a notification, your push application wakes up the Immich application on your phone to process the notification.

### Platform

- [X] Server
- [ ] Web
- [X] Mobile

**Comment by Thinkscape** [comment]
Some more context and discussion in #1671

In the meantime, there's this hack: https://github.com/pman07/Immich_Notify

**Comment by kingp0dd** [comment]
I thought i read somewhere that native app notifications are supported now. Or i remember incorrect

**Comment by jaggli** [comment]
This is really the last puzzle piece, after which I can finally ditch google photos, currently running the two in parallel, with some sync, juste beause I get notified, when others add pictures to shared albums

**Comment by kadledav** [comment]
This would be great to have natively — until then, I put together a small self-hosted workaround that does it via ntfy: [kadledav/immich-user-notify](https://github.com/kadledav/immich-user-notify)

It polls the API and sends a per-user ntfy push when photos or members are added to a shared album:
- one topic per person (derived from their email), not a single shared topic
- it doesn't ping the person who made the change
- auto-discovers all albums — no per-album configuration
- attributes changes ("Bob added new photos to…") and is localizable (currently EN/CS)

It's just a temporary fix until Immich supports this properly. Feedback welcome!
